### ISSUE TITLE
Remove /post/ from url

### ISSUE THREAD

**Issue by Kroellie** [mageplaza/magento-2-blog#195]
I noticed that the url is showing '/post/' in it. When I manually remove that, the url also works fine. Can it be removed by default?

So:
https://www.website.com/blog/post/article-1 --> https://www.website.com/blog/article-1

**Comment by wargis** [comment]
Can anyone provide any solution for the above issue?

**Comment by NgoDuVietHung97** [comment]
Hi,

Thanks for your response.

If you want to remove '/post/' by default in URL, you can download and replace the file I have attached.
Here is the path:

[Data.zip](https://github.com/mageplaza/magento-2-blog/files/3230372/Data.zip)
Data.php: app/code/Mageplaza/Helper/Data.php

If you have any other question, don't hesitate to contact me again.
Have a nice day!

**Comment by Kroellie** [comment]
Hi, thanks for your fix. Do you also commit this into your version, or should I keep these files somewhere myself?

**Comment by NgoDuVietHung97** [comment]
Hi @Kroellie,

We will update this in the next version of Blog Extension.
But for now, I think you should keep these files because the next version is in the finishing phase.
Thank you!

**Comment by mjoraid** [comment]
> Hi,
> 
> Thanks for your response.
> 
> If you want to remove '/post/' by default in URL, you can download and replace the file I have attached.
> Here is the path:
> 
> [Data.zip](https://github.com/mageplaza/magento-2-blog/files/3230372/Data.zip)
> Data.php: app/code/Mageplaza/Blog/Helper/Data.php
> 
> If you have any other question, don't hesitate to contact me again.
> Have a nice day!

What version is this compatible with?
I tried it on  "mageplaza/module-core": "^1.4.5""version": "3.1.1" and pages stopped loading, I quickly returned the original and the website loaded fine. 

Thanks. 

Best,

**Comment by mjoraid** [comment]
Update:

So I don't know what version the file **Data.php** is compatible with, I tried it with multiple version and it failed. 

**However, it seems the fix is only a small `if else` code:'**

1- open the file: path/to/extentions/Mageplaza/Blog/Helper/Data.php
2- go to function: `public function getBlogUrl($urlKey = null, $type = null, $store = null)` around line 560 
3- add the following code as the beginning of the function:

```
if ($type == "post") 
{
$type = "";
}
```

Basically it clears `$type` value if it contains the word `post `because it will be used in the URL generating. 

This worked for me on both version 2.2.9 and 2.3.5-p1.
